docs/design/include/mcmini/spy/intercept/wrappers.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef WRAPPERS_H
```

Comment:
This file also exists as `wrappers.h`, i.e. a C header. Which side is this intended for -- `libmcmini` or the `mcmini` process (or both)? If `libmcmini` it'd need to be a C header and the structs `MCTransition` etc. would have to change.

I think what you're looking for is `MCSharedTransition` as opposed to `MCTransition`. The former is the "real world"-side and the latter is strictly model side (it would map to `model::transition` now). That currently isn't clear though...